src/components/tab-panel/tab-panel.tsx

```diff
@@ -51,25 +51,36 @@ export class TabPanel {
 
     private slotElements: HTMLElement[] = [];
 
-    constructor() {
-        this.handleChangeTabs = this.handleChangeTabs.bind(this);
-        this.setSlotElements = this.setSlotElements.bind(this);
-        this.setTabStatus = this.setTabStatus.bind(this);
+    public connectedCallback() {
+        this.initialize();
     }
 
-    protected componentDidLoad() {
+    public componentDidLoad() {
+        this.initialize();
+    }
+
+    private initialize() {
         const slot = this.getSlot();
+        if (!slot) {
+            return;
+        }
+
         slot.addEventListener('slotchange', this.setSlotElements);
         this.setSlotElements();
         this.tabs.forEach(this.setTabStatus);
     }
 
-    // eslint-disable-next-line @stencil/own-methods-must-be-private
-    protected componentDidUnload() {
+    public disconnectedCallback() {
         const slot = this.getSlot();
         slot.removeEventListener('slotchange', this.setSlotElements);
     }
 
+    constructor() {
```

Comment:
I'm not sure we should move the constructor this far down in the component :thinking: 
We usually have it at the top everywhere else

Reply:
Hmm… I don't know how it ended up all the way down here. I agree, and I'll move it back up 👍

Reply:
Because there were two different commits that involved the same code, I couldn't move it with a fixup without creating a really messy conflict when rebasing it in the future. So did the change in an interactive rebase. The force-push I just did is only changing this one file.

Reply:
⚡